fdbserver/ConfigBroadcaster.actor.cpp

```diff
@@ -223,14 +224,26 @@ class ConfigBroadcasterImpl {
 			return Void();
 		}
 
+		// Clean up clientFailures futures. The values in this map should only
+		// be used for testing, and cleaning them up here ensures tests still
+		// have enough time to read the future while making sure the map
+		// doesn't grow unbounded.
+		for (auto it = impl->clientFailures.begin(); it != impl->clientFailures.end();) {
+			if (it->second.isReady()) {
+				it = impl->clientFailures.erase(it);
+			} else {
+				++it;
+			}
+		}
+
```

Comment:
Putting this O(n) loop inside every `registerWorker` call can get expensive. Instead, we can remove futures from `clientFailures` inside the `waitForFailure` callback.